### ISSUE TITLE
[UI] Apply title case to asset names, replace dashes with space

### ISSUE THREAD

**Issue by ckadner** [machine-learning-exchange/mlx#83]
Currently the UI displays the name of the assets (i.e. pipelines) as it was provided (or extracted from YAML) during pipeline registration. Depending on the means of registration (via MLX UI, KFP UI, KFP SDK, MLX API, MLX catalog import, or MLX bootstrapper) there are cases when the name defaults to a quasi ID like `trusted-ai` or `my-pipeline`. It may be useful to have the UI replace those dashes with spaces and apply title case to the name.

**Comment by drewbutlerbb4** [comment]
@ckadner How are these `quasi IDs` identified? Or are you suggesting that all dashes should be changed to spaces and title case should be forced on all names? This seems like a bandaid for a larger problem. Is there a reason we can't align the different MLX methods around the convention set by KFP?

In the meantime I can submit a PR to force show dashes as spaces and force title case.

**Comment by drewbutlerbb4** [comment]
@ckadner I have created a PR for this, if you or Animesh feel this is sufficient then feel free to close the issue.

**Comment by ckadner** [comment]
> @ckadner How are these `quasi IDs` identified? Or are you suggesting that all dashes should be changed to spaces and title case should be forced on all names?

I agree, there is a chance we might title-case something unwanted with a forced approach. Maybe we can filter for title/names that are all lower-case with dashes and not contain spaces, or, more strictly, the `name` equals the `id`. To avoid doing something unexpected, maybe the upload UI should force users to add a short given name, and validate that input based on a "title-case filter"?
